### PR TITLE
fix(tabs-extended): change anchor-based tabs to button-based tabs

### LINKED ISSUE #8357: [Tabs Extended]: Tabs that aren't open by default are unusable on iOS with VoiceOver
**Engineering info:**  
- Innovation Team engineer: @andy-blum 
- Innovation Team JIRA Ticket: https://jsw.ibm.com/browse/HC-2418
- DPO consulting engineer:

-----

### Description

When using iOS (iPad device) where `tabs-extended` is displayed with desktop design, with Voiceover, there is no way to change the active tab in the `tabs-extended` component. I think this may be related to using links in place of buttons, but regardless, the tabs component hides page content from screen reader users and prevents them from accessing content or links on hidden-by-default tabs.

For an example of how tabs should work per the W3, see [this codepen](https://codepen.io/andy-blum/full/KKyedzp)

### Component(s) impacted

tabs-extended, tabs-extended-with-media

### Browser

Safari

### Carbon for IBM.com version

Latest

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

carbon-for-ibmdotcom

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components/iframe.html?id=components-tabs-extended--default

### Steps to reproduce the issue (if applicable)

Attempt to use tabs on iOS with VO active

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

Fixes #8357 

### Description

Fixes an issue where iOS on tablet couldn't use tabs.

### Changelog

**Changed**

- changes `<a>` elements to `<button>` elements in tabs so iOS devices getting the desktop-style layout can navigate with voiceover

### CODE REVIEW

**Comment by RichKummer** [comment]
Hey @andy-blum , I tested this on iOS and I'm seeing that we're able to navigate between tabs with VoiceOver on. I noticed that when a tab is focused and if the tab has a tooltip, he focus area includes the tooltip even if the tab is not active yet. Is this the expected behavior for tabs with tooltips?

While Tab 1 is active but with Tab 2 selected, you can see the focus dips below the tab itself
IMG_1192

Here Tab 2 is active and you can see that the focus is capturing the tooltip
IMG_1193

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 1]
```diff
@@ -193,3 +209,64 @@
...
+  display: inline-block;
+  overflow: hidden;
+  width: calc(100% - 32px);
+  height: rem(40px);
+  padding: $spacing-04 0;
+  border-bottom: 1px solid $ui-03;
+  margin: 0 $spacing-05;
+  color: $text-02;
+  font-weight: 400;
+  line-height: 1rem;
```
Comment (kennylam) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:225]: Can use token here.

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 2]
```diff
@@ -193,3 +209,64 @@
...
+  transition: border $duration--fast-01 motion(standard, productive),
+    outline $duration--fast-01 motion(standard, productive);
+  white-space: nowrap;
+
+  &:focus,
+  &:active {
+    @include focus-outline('outline');
+
+    width: 100%;
+    padding-left: 16px;
```
Comment (kennylam) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:237]: Can use token here.

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 3]
```diff
@@ -64,11 +64,30 @@
         height: auto;
         overflow: visible;
         white-space: normal;
+        background: none;
+        border: initial;
+        border-bottom: 2px solid var(--cds-ui-03, #e0e0e0);
+        display: inline-flex;
+        width: 10rem;
```
Comment (kennylam) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:71]:
```suggestion
        width: rem(160px);
```

**Review by kennylam — COMMENTED** [review]
Thanks @andy-blum! Just a few small things.

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss (hunk 1 above)
Comment (kennylam) [reply packages/styles/scss/components/tabs-extended/_tabs-extended.scss:225]: https://www.carbondesignsystem.com/guidelines/spacing/overview/

**Comment by RichKummer** [comment]
Hey @andy-blum , I'll wait to re-review until Kenny's suggested changes are in, but I checked again on tablet with VO and it looks like the change to the tab selection was resolved - yay! 

I'll keep an eye out for the other changes and then run through it again then

**Review by RichKummer — APPROVED** [review]
Looks great @andy-blum , thanks for making the change!

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 4]
```diff
@@ -64,11 +64,30 @@
         height: auto;
         overflow: visible;
         white-space: normal;
+        background: none;
+        border: initial;
+        border-bottom: 2px solid var(--cds-ui-03, #e0e0e0);
```
Comment (proeung) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:69]:
```suggestion
        border-bottom: carbon--rem(2px) solid $ui-03;
```

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 5]
```diff
@@ -193,3 +209,64 @@
 @include exports('tabs-extended') {
   @include tabs-extended;
 }
+
+button.#{$prefix}--tabs__nav-link {
+  @include focus-outline('reset');
+
+  display: inline-block;
+  overflow: hidden;
+  width: calc(100% - 32px);
```
Comment (proeung) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:218]:
```suggestion
  width: calc(100% - #{$spacing-07});
```

File: packages/styles/scss/components/tabs-extended/_tabs-extended.scss [hunk 6]
```diff
@@ -193,3 +209,64 @@
...
+    width: rem(160px);
+    padding: $spacing-04 $spacing-05 $spacing-03;
+    border-bottom: $tab-underline-color;
+    margin: 0;
+    line-height: inherit;
+
+    &:focus,
+    &:active {
+      width: rem(160px);
+      border-bottom: 2px;
```
Comment (proeung) [inline packages/styles/scss/components/tabs-extended/_tabs-extended.scss:251]:
```suggestion
      border-bottom: carbon--rem(2px);
```

**Review by proeung — CHANGES_REQUESTED** [review]
Looks good overall! I just have a few suggestions related to keeping with using carbon tokens for color and spacings.

**Comment by mwanberg** [comment]
I think this will need to be merged before I can finish my related PR: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8394

**Review by proeung — APPROVED** [review]
LGTM! Thanks for applying my suggestions!

**Pull request merged**